compiler/dyno/lib/resolution/resolution-queries.cpp

```diff
@@ -1441,6 +1453,56 @@ const TypedFnSignature* instantiateSignature(Context* context,
 
       formalIdx++;
     }
+
+    // At this point, we have computed the instantiated type for this
+    // formal. However, what we're still missing some information,
+    // and furthermore, we have not enforced type query constraints.
+
+    if (entry.isVarArgEntry()) {
+      // Vararg entries don't get substitutions at this point, so
+      // manually update type queries.
+      if (auto vld = formal->toVarLikeDecl()) {
+        if (auto te = vld->typeExpression()) {
+            visitor.resolveTypeQueries(te, useType.type());
+        }
+      }
+    } else {
+      // Substitutions have been updated; re-run resolution to get better
+      // intents, vararg info, and to extract type query info.
+      formal->traverse(visitor);
```

Comment:
I've lost track since our last discussion: what would happen if we set 'ignoreSubstitutionsFor' in this traversal?